src/System.Private.CoreLib/shared/System/SpanHelpers.Char.cs

```diff
@@ -175,6 +175,206 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
             }
         }
 
+        public static unsafe int IndexOfAny(ref char searchSpace, char value0, char value1, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            fixed (char* pChars = &searchSpace)
+            {
+                char* pCh = pChars;
+                char* pEndCh = pCh + length;
+
+#if !netstandard11
+                if (Vector.IsHardwareAccelerated && length >= Vector<ushort>.Count * 2)
```

Comment:
Out of scope for this PR, but it might be nice to try this with Hardware Intrinsics as well.